Issue #16262: do not override generic format if not reserved and only printing (#16329)
coq/coq: doc/changelog/03-notations/16329-master+experiment-addressing16262-heuristic-redefining-notation-format.rst

- **Changed:**

>   an :g:`only printing` interpretation of a notation with a specific
>   format does not change any longer the printing rule of other
>   interpretations of the notation; to globally change the default
>   printing rule of all interpretations of a notation, use instead
>   :g:`Reserved Notation`

Reviewer: ```suggestion
  an :g:`only printing` interpretation of a notation with a specific
  format no longer changes the printing rule of other
  interpretations of the notation.  To globally change the default
  printing rule of all interpretations of a notation, use
  :g:`Reserved Notation` instead
```

I couldn't find a definition of "interpretation" in the syntax extensions chapter.  If you're willing, perhaps you could add one?  (And maybe add a glossary entry under "notation interpretation" since "interpretation" is such a generic term.)

Also, I don't see a clear definition of "format", either, which IIUC is a string that includes extra info on where to break lines.

Author: The interpretation is the "right-hand side". It is a common terminology and we should indeed add it to the manual (e.g. in "A notation binds a syntactic expression to a term, called its interpretation", and adding "interpretation" to the index; and the sentence "The right-hand side of a notation is interpreted at the time the notation is given. In particular" should be removed from the note which follows).

> Also, I don't see a clear definition of "format", either, which IIUC is a string that includes extra info on where to break lines.

Indeed. Then, for "format" in the manual, maybe could we rephrase "A format is an extension of the string ..." into "A format tells how to print a notation. It is a string extending the notation string with the following possible elements..." + putting format in the index? (Or "to nicely print", or "to print a notation nicely formatted", ...?)
